### PR TITLE
Added the agg argument to EnsembleForecaster

### PR DESCRIPTION
**Pull request by Ifeanyi30**
#### Reference Issues/PRs

fixes the issue #583

#### What does this implement/fix? Explain your changes.

The agg argument adds aggregate options for ensembling forecasters.
#### Does your contribution introduce a new dependency? If yes, which one?
No

#### What should a reviewer concentrate their feedback on?

The changes on the __init__ of the EnsembleForecaster and it predict method.
##### For all contributions
- [ ] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [ ] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [ ] I've added unit tests and made sure they pass locally.

##### For new estimators
- [ ] I've added the estimator to the online documentation.
- [ ] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

File: sktime/forecasting/compose/_ensemble.py [hunk 1]
```diff
@@ -28,8 +28,9 @@ class EnsembleForecaster(
 
     _required_parameters = ["forecasters"]
 
-    def __init__(self, forecasters, n_jobs=None):
+    def __init__(self, forecasters, n_jobs=None, agg="mean"):
```
Comment (mloning) [inline sktime/forecasting/compose/_ensemble.py:31]: We need to add this to the docstring too.

File: sktime/forecasting/compose/_ensemble.py [hunk 2]
```diff
@@ -73,6 +74,29 @@ def update(self, y, X=None, update_params=True):
...
+        return_pred_int : boolean, optional (default=False)
+        alpha : fh : float, (default=DEFAULT_ALPHA)
+
+        Returns
+        -------
+        DataFrame : aggregate dataframe reduction.
+        """
         if return_pred_int:
             raise NotImplementedError()
+        if self.agg not in ("median", "mean", "min", "max"):
```
Comment (mloning) [inline sktime/forecasting/compose/_ensemble.py:92]: I would write something like this
```suggestion
        valid_agg_funcs = ("median", "mean", "min", "max")
        if self.agg not in valid_agg_funcs:
```
You can then reuse `valid_agg_funcs` in the error message.

File: sktime/forecasting/compose/_ensemble.py (hunk 1 above)
Comment (mloning) [reply sktime/forecasting/compose/_ensemble.py:31]: I'd call `agg` `aggfunc` perhaps, this seems to be common name in pandas.

File: sktime/forecasting/compose/_ensemble.py [hunk 3]
```diff
@@ -73,6 +74,29 @@ def update(self, y, X=None, update_params=True):
...
+            raise ValueError(
+                "Invalid agg value. Valid values are 'median', 'mean', 'min', 'max'"
+            )
+        if self.agg == "median":
```
Comment (mloning) [inline sktime/forecasting/compose/_ensemble.py:96]: Can we not first compute `y_pred = pd.concat(self._predict_forecasters(fh, X), axis=1)` and then aggregate it? Save us some code duplication.

**Review by mloning — COMMENTED** [review]
Thanks @Ifeanyi30 - please find below a few minor comments.

**Comment by Ifeanyi30** [comment]
Copy that

**Comment by Ifeanyi30** [comment]
Just made the corrections

File: sktime/forecasting/compose/_ensemble.py [hunk 4]
```diff
@@ -87,16 +87,18 @@ def _predict(self, fh, X=None, return_pred_int=False, alpha=DEFAULT_ALPHA):
         -------
         DataFrame : aggregate dataframe reduction.
         """
+        valid_agg_funcs = ("median", "mean", "min", "max")
+        d_frame = pd.concat(self._predict_forecasters(fh, X), axis=1)
```
Comment (mloning) [inline sktime/forecasting/compose/_ensemble.py:92]: I would call `d_frame` `y_pred` instead or `Y_pred` if you want to indicate that it's not just a single prediction.

**Comment by Ifeanyi30** [comment]
Pls can you review the changes I made? Also can I add the to the outreachy
contribution list for this project's final application?

**Comment by mloning** [comment]
Hi @Ifeanyi30, I opened a PR at https://github.com/Ifeanyi30/sktime/pull/1 with some minor suggestions. If you agree with them, please merge it. Those are all of my suggestions and this PR should be ready after that.

**Comment by Ifeanyi30** [comment]
Just merged the PR with mine. I do have another PR I'd like you to review "
https://github.com/alan-turing-institute/sktime/pull/776"

**Review by mloning — APPROVED** [review]

**Comment by mloning** [comment]
Thanks @Ifeanyi30 - now merged! 🎉

**Comment by Ifeanyi30** [comment]
Awesome, thanks!

**Pull request merged**